### ISSUE TITLE
Get inf ctc loss during training

### ISSUE THREAD

**Issue by tz301** [k2-fsa/icefall#135]
I'm using my labeled data for training and get inf ctc loss.

I have debug it, and find the batch which changes ctc loss to inf has very short data. The minimum of number frames (before subsampling) is 11. 

I try to generate train manifest which filter short segment less than 1s (similiar as kaldi). It seems to be ok.

So maybe this very short data cause inf loss?
If so, maybe this can be shown in some code or tutorial? And what is the recommended minimum threshold for this value?

**Comment by csukuangfj** [comment]
> The minimum of number frames (before subsampling) is 11.

If you are using conformer_ctc, which has a subsampling factor of 4, you will get `( (11 - 1) / 2 - 1) / 2 = 2` frames.
One of the assumptions of CTC is that the output label length is not greater than that of the input length.
If that assumption does not hold, you will get an empty lattice, and hence an `inf` value for the total score.

> And what is the recommended minimum threshold for this value?

You can do one of the following:
- (1) Exclude `inf` values from ctc loss
https://github.com/k2-fsa/icefall/blob/243fb9723cb82287ec5a891155ab9e0bc304740d/egs/librispeech/ASR/conformer_ctc/train.py#L367-L373

(Note you have to first set `params.reduction` to `none`, filter `inf` values from `ctc_loss`, and then compute the sum.

- (2) Filter utterances that are too short from the manifest.

**Comment by tz301** [comment]
Thanks, I will try to simply filter short utterances and adjust some threshold for experiments.

**Comment by AmirHussein96** [comment]
My solution: 
ctc_loss = k2.ctc_loss(decoding_graph=decoding_graph, 
dense_fsa_vec=dense_fsa_vec,
output_beam=params.beam_size,
reduction='none',
use_double_scores=params.use_double_scores,
    )
# filter inf from ctc_loss
ctc_loss = torch.sum(torch.where(ctc_loss != float('inf'),ctc_loss , torch.tensor(0, dtype=torch.float32).to(device)))

**Comment by csukuangfj** [comment]
By the way, you can replace `ctc_loss != float('inf')` with
```python3
torch.isinf(ctc_loss)
```
